server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -69,6 +70,9 @@
     public static final Setting<Translog.Durability> INDEX_TRANSLOG_DURABILITY_SETTING =
         new Setting<>("index.translog.durability", Translog.Durability.REQUEST.name(),
             (value) -> Translog.Durability.valueOf(value.toUpperCase(Locale.ROOT)), Property.Dynamic, Property.IndexScope);
+    public static final Setting<Integer> TRANSLOG_ASYNC_EVENT_BUFFER_SIZE =
+        Setting.intSetting("indices.translog.async_event_buffer_size", 256 * 1026, Property.NodeScope);
```

Comment:
This would consume a significant amount of memory.

Reply:
this can be set a smaller size.
But using memory to buffer translog when writing is unavailable  is the key point of this proposal, it's a trade off between index speed and memory.